### PR TITLE
SALTO-5534 - Salesforce: support data records deletions in quickfetch

### PR DESCRIPTION
**Pull request by tomsellek**
In the quickfetch case we need to determine which records were deleted on the Salesforce side so we can delete them on our end. The simple way to do this would be to use a SOQL query with all the IDs of the records that exist in the workspace and see which of them are not returned. 

---

There's a small optimization of not querying for IDs of records that we just fetched, because we know those records were not deleted.

Tests:
- Do a full fetch, delete an account, do a quickfetch, observe the account being deleted.

---
_Release Notes_: 
Salesforce: quickfetch will now correctly identify deleted records

---
_User Notifications_: 
N/A

### CODE REVIEW

File: packages/salesforce-adapter/src/filters/custom_objects_instances.ts [hunk 1]
```diff
@@ -863,6 +900,24 @@ const filterCreator: RemoteFilterCreator = ({ client, config }) => ({
...
+          instances.map(instanceInternalId),
+        ),
+      )
+      _.remove(
+        elements,
+        (element) =>
+          isInstanceOfCustomObjectSync(element) &&
+          !idsInSalesforce.has(instanceInternalId(element)),
+      )
+    }
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/filters/custom_objects_instances.ts:919]: Will this code actually delete Records that were not changed?
The `elements` list will only contain fetched Elements.. No?

I think this logic needs to sit next to where we detected deletions for Metadata (in `adapter.ts` fetch)

**Review by tamtamirr — COMMENTED** [review]
Added a comment. I might need a sync on this PR 🙏

File: packages/salesforce-adapter/src/filters/custom_objects_instances.ts (hunk 1 above)
Comment (tomsellek) [reply packages/salesforce-adapter/src/filters/custom_objects_instances.ts:919]: Oh. Yeah. Right. Crap.

Comment (tomsellek) [reply packages/salesforce-adapter/src/filters/custom_objects_instances.ts:919]: Why not add removal changes in the filter's `onDeploy`?

File: packages/salesforce-adapter/src/adapter.ts [hunk 2]
```diff
@@ -973,4 +975,82 @@ export default class SalesforceAdapter implements AdapterOperations {
...
+      await querySalesforceForRecordIdsOfInstances(workspaceInstances),
+    )
+
+    return _(workspaceInstances)
+      .filter(
+        (instance) =>
+          !instanceIdsInSalesforce.has(instanceInternalId(instance)),
+      )
+      .map((instance) => instance.elemID)
+      .value()
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/adapter.ts:1038]: nit: Lodash is not required here..?

File: packages/salesforce-adapter/test/adapter.discover.test.ts [hunk 3]
```diff
@@ -3073,3 +3085,152 @@ describe('Fetch via retrieve API', () => {
...
+      it('should return the modified instance', () => {
+        expect(fetchResult.elements).toContainEqual(
+          expect.objectContaining({
+            elemID: testInstance.elemID,
+          }),
+        )
+      })
+    })
+  })
+})
```
Comment (tamtamirr) [inline packages/salesforce-adapter/test/adapter.discover.test.ts:3236]: What is this addition?
I would also expect the test of deleted data records to be within `partial fetch deletion support` describe.
Please also note: This support you've added is also relevant for targeted fetch (partial)

**Review by tamtamirr — COMMENTED** [review]
LGTM! Added some comments

File: packages/salesforce-adapter/src/adapter.ts (hunk 2 above)
Comment (tomsellek) [reply packages/salesforce-adapter/src/adapter.ts:1038]: Ack. Must be leftovers.

File: packages/salesforce-adapter/test/adapter.discover.test.ts (hunk 3 above)
Comment (tomsellek) [reply packages/salesforce-adapter/test/adapter.discover.test.ts:3236]: > What is this addition?

Unit tests for the new code?

> I would also expect the test of deleted data records to be within partial fetch deletion support describe.

Ack

> Please also note: This support you've added is also relevant for targeted fetch (partial)

Ack.

**Review by tamtamirr — APPROVED** [review]
LGTM!

**Pull request merged**